src/Plugin/Condition/PathHasAlias.php

```diff
@@ -72,11 +73,18 @@ public static function create(ContainerInterface $container, array $configuratio
   }
 
   /**
-   * {@inheritdoc}
+   * Check if path has alias.
+   *
+   * @param string $path
+   *   The path to check.
+   * @param \Drupal\Core\Language\LanguageInterface|null $language
+   *   The language to check for a path in or NULL.
+   *
+   * @return bool
+   *   TRUE if the path has an alias in the given language.
    */
-  public function evaluate() {
-    $path = $this->getContextValue('path');
-    $language = $this->getContext('language')->hasContextValue() ? $this->getContextValue('language')->getId() : NULL;
+  protected function doEvaluate($path, LanguageInterface $language = NULL) {
+    $language = is_null($language) ? NULL : $language->getId();
```

Comment:
Can we name this variable $langcode, same as on AliasManagerInterface? It is confusing to overwrite the parameter variable.